### ISSUE TITLE
Matrix plan_transform

### ISSUE THREAD

**Issue by ioannisPApapadopoulos** [JuliaApproximation/PiecewiseOrthogonalPolynomials.jl#79]
Following on from my issue in `ClassicalOrthogonalPolynomials`.. 

My goal is to apply a tensor-product `plan_transform(ContinuousPolynomial{0}(r), (Block(p), Block(p))` to multiple ``vectors`` simultaneously. Focusing on the 1D case first, it seems to me that the infrastructure is not completely there yet, right?

From the tests I can see that the following works:
```
r, p = range(0,1,3), 3
P = ContinuousPolynomial{0}(r)
x, F = plan_grid_transform(P, (Block(p),2), 1)
c = F*[exp.(x) ;;; cos.(x)]
```
but then `F \ c` errors. 

Do you think the goal is attainable @dlfivefifty ?

**Comment by dlfivefifty** [comment]
What error do you get?

The entire infrastructure was designed with this in mind. As you note since the grid is a matrix so if you have multiple functions its a 3-tensor. So this should work.... probably I just never got around to finishing the implementation.

**Comment by ioannisPApapadopoulos** [comment]
Yes, I think the implementation is not finished for `inv_perm_blockvec`. The error is:

```
ERROR: AssertionError: dims == 1:2 || dims == (1, 2)
Stacktrace:
 [1] _inv_perm_blockvec(X::BlockArrays.BlockedMatrix{Float64, Matrix{…}, Tuple{…}}, dims::Int64)
   @ PiecewiseOrthogonalPolynomials C:\Users\john.papad\.julia\packages\PiecewiseOrthogonalPolynomials\NDNMh\src\piecewisepolynomial.jl:104
 [2] \(F::PiecewiseOrthogonalPolynomials.ApplyPlan{…}, X::BlockArrays.BlockedMatrix{…})
   @ PiecewiseOrthogonalPolynomials C:\Users\john.papad\.julia\packages\PiecewiseOrthogonalPolynomials\NDNMh\src\piecewisepolynomial.jl:114
 [3] top-level scope
   @ c:\Users\john.papad\.julia\dev\HierarchicalObstacles\experiments\tmp_parallel.jl:83
Some type information was truncated. Use `show(err)` to see complete types.
```

**Comment by dlfivefifty** [comment]
Right. That assertion is just a message to myself saying I haven't completed it.

**Comment by ioannisPApapadopoulos** [comment]
resolved with 333b297a147da84e42c6a3d955ac9794d3bfb693
